frontend/public/components/nav/navSortUtils.ts

```diff
@@ -109,5 +107,39 @@ export const sortExtensionItems = (
     }, []);
   };
 
-  return extensionItems.sort((a, b) => dependencies(b.properties.id).indexOf(a.properties.id) * -1);
+  const sortItems = (preSorted, itemsToSort) => {
+    if (itemsToSort.length < 2) {
+      preSorted.push(...itemsToSort);
+      return;
+    }
+
+    let sortedItem = false;
+    const remainingItems = [];
+    itemsToSort.forEach((item) => {
+      const deps = dependencies(item.properties.id);
+      // If not dependant on any items to be sorted, ok to add it in
+      if (!deps.find((id) => itemsToSort.find((i) => i.properties.id === id))) {
+        sortedItem = true;
+        preSorted.push(item);
+      } else {
+        // Still has a dependency
+        remainingItems.push(item);
+      }
+    });
+
+    if (remainingItems.length) {
+      // If nothing changed, just add the remaining items
+      if (!sortedItem) {
+        preSorted.push(...remainingItems);
+        return;
+      }
```

Comment:
Is this an optimization? Maybe its me, but I don't get for which case this is needed. At least the tests works fine if I remove this 5 lines. :)

Reply:
If there are no items left that have dependencies on other items then we add them all. This is for error cases where two items might depend on each other.

Reply:
I've added a test for such thing

Reply:
Yeah great. Without this code we get an endloss loop. Thanks for that additional test. Works fine! :+1: